components/infobox/wikis/rainbowsix/infobox_team_custom.lua

```diff
@@ -0,0 +1,105 @@
+---
+-- @Liquipedia
+-- wiki=rainbowsix
+-- page=Module:Infobox/Team/Custom
+--
+-- Please see https://github.com/Liquipedia/Lua-Modules to contribute
+--
+
+local Team = require('Module:Infobox/Team')
+local Earnings = require('Module:Earnings of')
+local Variables = require('Module:Variables')
+local Flags = require('Module:Flags')
+local Class = require('Module:Class')
+local Injector = require('Module:Infobox/Widget/Injector')
+local Cell = require('Module:Infobox/Widget/Cell')
+local String = require('Module:String')
+local Template = require('Module:Template')
+
+local CustomTeam = Class.new()
+
+local _CURRENT_YEAR = os.date('%Y')
+local _START_YEAR = 2015
+
+local CustomInjector = Class.new(Injector)
+
+local _team
+
+function CustomTeam.run(frame)
+	local team = Team(frame)
+	_team = team
+	team.createBottomContent = CustomTeam.createBottomContent
+	team.addToLpdb = CustomTeam.addToLpdb
+	team.defineCustomPageVariables = CustomTeam.defineCustomPageVariables
+	team.createWidgetInjector = CustomTeam.createWidgetInjector
+	return team:createInfobox(frame)
+end
+
+function CustomInjector:parse(id, widgets)
+	if id == 'earnings' then
+		local earnings = Earnings.team({_team.args.id or _team.pagename})
+		Variables.varDefine('earnings', earnings:gsub(',', ''))
+		if earnings == 0 then
+			earnings = nil
+		else
+			earnings = '$' .. earnings
+		end
+		return {
+			Cell{
+				name = 'Earnings',
+				content = {
+					earnings
+				}
+			}
+		}
+	elseif id == 'region' then
+		return {
+			Cell{name = 'Region', content = {_team:_createRegion(_team.args.region, _team.args.location)}}
+		}
+	end
+	return widgets
+end
+
+function CustomTeam:createBottomContent()
+	return Template.safeExpand(
+		mw.getCurrentFrame(),
+		'Upcoming and ongoing matches of',
+		{team = _team.pagename or _team.name},
+		''
+	)
+end
+
+function CustomTeam:addToLpdb(lpdbData, args)
+	lpdbData.earnings = Variables.varDefault('earnings', 0)
+	if not String.isEmpty(args.teamcardimage) then
+		lpdbData.logo = 'File:' .. args.teamcardimage
+	elseif not String.isEmpty(args.image) then
+		lpdbData.logo = 'File:' .. args.image
+	end
```

Comment:
This seems like the thing we want to enforce equally across all wikis, _if we do it in the first place_. Please remove this code. If R6 desires to have this, it should be brought up as discussion. The current thinking is instead to add a second property to team LPDB that saves the `teamcardimage`

Reply:
We could bot change all `{{Infobox team|...|teamcardimage=foo.bar|...}}` into `{{TeamCardImage|startdate=|enddate=|filename=foo.bar}} {{Infobox team|...|...}}`? 
This is the way to set up when a team have multiple TeamCard Images (historic images). This would drop the need of handling in the infobox template. 
Adding a property in lpdb_team for `teamcardimage` I don't think is optimal due teams can have historic TeamCardImages. Additionally, if I'm not mistaken, `teamcardimage` is/should be deprecated in favor of `TeamTemplate's image`, no?

Reply:
I have no real opinion here, all I'm saying is that we need to add a property to LPDB that contains the team logo without the text.

Reply:
from wiki perspective we can always use teamtemplates for that (with the new file standards)
i assume you need it for the API, right?
if so maybe it would be good to add a function to the commons module that fetches the teams current icon from team templates and stores that into lpdb (after the new field exists)

Reply:
> i assume you need it for the API, right?

Correct

Reply:
Based on discussions on discord, a new field will be added. Name of the field is TBD. This field should always contain the textless logo, while image may contain a logo with text.

Reply:
@JaspervanRiet How about closing this comment & open an issue. Once the LDPB field has been added we'll make a PR for it to standardize it across all wikis.

Reply:
imo issue is good
needs to be adjusted in existing implementations anyways

Reply:
#776